lib/ansible/inventory/manager.py

```diff
@@ -276,6 +279,10 @@ def parse_source(self, source, cache=False):
                     except Exception as e:
                         display.debug('%s failed to parse %s' % (plugin_name, to_text(source)))
                         failures.append({'src': source, 'plugin': plugin_name, 'exc': AnsibleError(e)})
+                    finally:
+                        if not parsed:
+                            # restore inventory to avoid partial/incomplete updates
+                            self._inventory = inventory
```

Comment:
FWIW, testing locally, and it seems the restore is somehow affecting groups-of-groups, essentially not loading them.  Running with debug and vvv on, but I'm not seeing any obvious reason.  If I remove the `finally` block, groups-of-groups work fine.

Reply:
deserialization issue, this has uncovered a flaw on how inventory objects interact